packages/invoices/src/OutboundShipment/DetailView/modals/ItemDetailsModal.tsx

```diff
@@ -68,56 +75,88 @@ const sortByDisabledThenExpiryDate = (a: BatchRow, b: BatchRow) => {
   return 0;
 };
 
+const useStockLines = (itemCode: string | undefined) => {
+  const [batchRows, setBatchRows] = useState<BatchRow[]>([]);
+  const { data, isLoading } = useItems({
+    code: { equalTo: itemCode },
+  });
+
+  useEffect(() => {
+    if (!itemCode) return;
+    data?.onFilterByCode(itemCode);
+  }, [itemCode]);
+
+  useEffect(() => {
+    if (!data) return;
+
+    const { nodes } = data;
+    const { availableBatches } = nodes[0] ?? { availableBatches: [] };
+
+    setBatchRows(() => {
+      return availableBatches
+        .map(batch => ({
+          ...batch,
+          quantity: 0,
+        }))
+        .sort(sortByDisabledThenExpiryDate);
+    });
+  }, [data]);
+
+  const onIssue = (stockLineId: string, quantity: number) => {
+    const batchRowIdx = batchRows.findIndex(batch => batch.id === stockLineId);
+    const batchRow = batchRows[batchRowIdx];
+    if (!batchRow) return;
+
+    batchRows[batchRowIdx] = { ...batchRow, quantity };
+    setBatchRows([...batchRows]);
+  };
+
+  return { batchRows, onIssue, isLoading };
+};
+
 export const ItemDetailsModal: React.FC<ItemDetailsModalProps> = ({
   invoiceLine,
   isOpen,
   onClose,
   upsertInvoiceLine,
+  onChangeItem,
+  summaryItem,
 }) => {
-  const [batchRows, setBatchRows] = React.useState<BatchRow[]>([]);
-  const [selectedItem, setSelectedItem] = React.useState<Item | null>(null);
-  const [quantity, setQuantity] = React.useState(0);
-  const [allocated, setAllocated] = React.useState(0);
-  const [packSize, setPackSize] = React.useState(1);
+  const [batchRows, setBatchRows] = useState<BatchRow[]>([]);
+
+  const [quantity, setQuantity] = useState(0);
+  const [allocated, setAllocated] = useState(0);
+  const [packSize, setPackSize] = useState(1);
+
+  const methods = useForm({ mode: 'onBlur' });
+  const { reset, register, setValue, getValues } = methods;
```

Comment:
So, `selectedItem` I lifted up the the `DetailView` and made an `OutboundSummaryItem` and `batchRows` lifted out to `useStockLines`.